model-optimizer/extensions/ops/gather.py

```diff
@@ -17,19 +18,36 @@ def __init__(self, graph: Graph, attrs: dict):
         super().__init__(graph, {
             'op': self.op,
             'type': self.op,
-            'version': 'opset1',
-
+            'version': 'opset7',
+            'batch_dims': 0,
             'infer': self.infer,
-
             'force_precision_in_ports': {1: 'int32', 2: 'int64'},
-
             'in_ports_count': 3,
             'out_ports_count': 1,
         }, attrs)
 
         assert 'axis' not in self.attrs, \
             'Use AttributedGather operation instead of Gather to create it with `axis` as a parameter'
 
+    def backend_attrs(self):
+        version = self.get_opset()
+        if version == 'opset7':
+            return [('batch_dims', lambda node: Gather.get_batch_dims(node))]
+        elif version == 'opset1':
+            return []
+        else:
+            raise Error('Unsupported operation opset version "{}"'.format(version))
+
+    @staticmethod
+    def get_axis(node: Node):
+        data_rank = len(node.in_port(0).data.get_shape())
+        axis_value = node.in_port(2).data.get_value()
+        return axis_value + data_rank if axis_value < 0 else axis_value
+
+    @staticmethod
+    def get_batch_dims(node: Node):
+        return node.batch_dims + Gather.get_axis(node) if node.batch_dims < 0 else node.batch_dims
```

Comment:
We should have allowed negative values for `batch_dims` attribute in the spec. Because current approach is not dynamism friendly. Let's discuss this separately and this does not prevent us from merging this PR.

Reply:
let's discuss this separately.